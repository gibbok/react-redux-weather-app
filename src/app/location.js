// @flow
import {LocationType} from '../types'  // eslint-disable-line no-unused-vars

const location = ():LocationType => {
  return {
    locationId: 5128581, // nyc usa
    geo: [-74.006, 40.7143],
    isFetching: false
  }
}

export default location
