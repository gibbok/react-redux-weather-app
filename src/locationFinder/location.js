const Location = ({onClick, name, country}) => (
  <li onClick={onClick}>
    {name} {country}
  </li>
)

export default Location
