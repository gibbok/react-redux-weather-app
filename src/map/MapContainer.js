import { connect } from 'react-redux'
import Map from './Map'
import store from '../app/store'

let isInitiated = false

const map = (map) => {
  return map
}

// const getMapLocationId = () => {
//   return store.getState().mapReducer.map.app.locationId
// }

const onInit = () => {
  if (isInitiated) {
    return
  }
  isInitiated = true
  // store.dispatch(getWeather(getWeatherLocationId()))
}

const mapStateToProps = (state) => {
  return {
    map: map(state.mapReducer.map.data),
    onInit: onInit()
  }
}

const MapContainer = connect(
  mapStateToProps
)(Map)

export default MapContainer
