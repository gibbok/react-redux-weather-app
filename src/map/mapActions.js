// @flow
import {ActionType} from '../../types' // eslint-disable-line no-unused-vars

import * as types from './mapActionTypes'

/*
 * Action set a region.
 */
const setMapRegion = (value:string):ActionType => ({
  type: types.SET_MAP_REGION,
  payload: value
})

/*
 * Action set a map type.
 */
const setMapType = (value:string):ActionType => ({
  type: types.SET_MAP_TYPE,
  payload: value
})

export {setMapRegion, setMapType}
