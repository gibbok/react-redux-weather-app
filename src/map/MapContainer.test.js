// @flow
import React from 'react'
import { createMockStore, createMockDispatch } from 'redux-test-utils'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import MapContainer from './MapContainer'
import { setMapRegion, setMapType } from './mapActions'
import location from '../app/location'

describe('WeatherContainer', () => {
  it('should render', () => {
    const initialState = {
      mapReducer: {
        map: {
          data: {
            regions: [
              {
                id: 'currentLocation',
                title: 'Current Location',
                isActive: true
              },
              {
                id: 'northAmerica',
                title: 'North America',
                isActive: false
              },
              {
                id: 'southAmerica',
                title: 'South America',
                isActive: false
              },
              {
                id: 'europe',
                title: 'Europe',
                isActive: false
              },
              {
                id: 'asia',
                title: 'Asia',
                isActive: false
              },
              {
                id: 'pacificIslands',
                title: 'Pacific Islands',
                isActive: false
              },
              {
                id: 'africa',
                title: 'Africa',
                isActive: false
              }
            ],
            types: [
              {
                id: 'temperature',
                title: 'Temperature',
                isActive: true
              },
              {
                id: 'precipitation',
                title: 'Precipitation',
                isActive: false
              },
              {
                id: 'pressure',
                title: 'Pressure',
                isActive: false
              },
              {
                id: 'wind',
                title: 'Wind',
                isActive: false
              },
              {
                id: 'cloud',
                title: 'Cloud',
                isActive: false
              }
            ]
          },
          app: {
            locationId: location().locationId,
            geo: location().geo,
            isFetching: false
          },
          ui: {
            type: 'temperature',
            region: 'northAmerica'
          }
        }
      }
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<MapContainer store={store} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should dispatch action setMapRegion', () => {
    const state = 'state'
    const store = createMockStore(state)
    const action = setMapRegion('europe')

    store.dispatch(action)
    expect(store.getAction(action.type)).toEqual(action)
    expect(store.getActions()).toEqual([action])
    expect(store.isActionDispatched(action)).toBe(true)
    expect(store.isActionTypeDispatched(action.type)).toBe(true)
    expect(store.getState()).toBe(state)

    const dispatchMock = createMockDispatch()
    dispatchMock.dispatch(action)
    expect(dispatchMock.getAction(action.type)).toEqual(action)
    expect(dispatchMock.getActions()).toEqual([action])
    expect(dispatchMock.isActionDispatched(action)).toBe(true)
    expect(dispatchMock.isActionTypeDispatched(action.type)).toBe(true)
  })

  it('should dispatch action setMapType', () => {
    const state = 'state'
    const store = createMockStore(state)
    const action = setMapType('wind')

    store.dispatch(action)
    expect(store.getAction(action.type)).toEqual(action)
    expect(store.getActions()).toEqual([action])
    expect(store.isActionDispatched(action)).toBe(true)
    expect(store.isActionTypeDispatched(action.type)).toBe(true)
    expect(store.getState()).toBe(state)

    const dispatchMock = createMockDispatch()
    dispatchMock.dispatch(action)
    expect(dispatchMock.getAction(action.type)).toEqual(action)
    expect(dispatchMock.getActions()).toEqual([action])
    expect(dispatchMock.isActionDispatched(action)).toBe(true)
    expect(dispatchMock.isActionTypeDispatched(action.type)).toBe(true)
  })
})
