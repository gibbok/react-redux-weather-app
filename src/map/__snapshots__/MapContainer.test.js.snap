// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherContainer should render 1`] = `
<Map
  geo={
    Array [
      -74.006,
      40.7143,
    ]
  }
  map={
    Object {
      "regions": Array [
        Object {
          "id": "currentLocation",
          "isActive": true,
          "title": "Current Location",
        },
        Object {
          "id": "northAmerica",
          "isActive": false,
          "title": "North America",
        },
        Object {
          "id": "southAmerica",
          "isActive": false,
          "title": "South America",
        },
        Object {
          "id": "europe",
          "isActive": false,
          "title": "Europe",
        },
        Object {
          "id": "asia",
          "isActive": false,
          "title": "Asia",
        },
        Object {
          "id": "pacificIslands",
          "isActive": false,
          "title": "Pacific Islands",
        },
        Object {
          "id": "africa",
          "isActive": false,
          "title": "Africa",
        },
      ],
      "types": Array [
        Object {
          "id": "temperature",
          "isActive": true,
          "title": "Temperature",
        },
        Object {
          "id": "precipitation",
          "isActive": false,
          "title": "Precipitation",
        },
        Object {
          "id": "pressure",
          "isActive": false,
          "title": "Pressure",
        },
        Object {
          "id": "wind",
          "isActive": false,
          "title": "Wind",
        },
        Object {
          "id": "cloud",
          "isActive": false,
          "title": "Cloud",
        },
      ],
    }
  }
  onRegionsChange={[Function]}
  onTypesChange={[Function]}
  regions={
    Array [
      Object {
        "id": "currentLocation",
        "isActive": true,
        "title": "Current Location",
      },
      Object {
        "id": "northAmerica",
        "isActive": false,
        "title": "North America",
      },
      Object {
        "id": "southAmerica",
        "isActive": false,
        "title": "South America",
      },
      Object {
        "id": "europe",
        "isActive": false,
        "title": "Europe",
      },
      Object {
        "id": "asia",
        "isActive": false,
        "title": "Asia",
      },
      Object {
        "id": "pacificIslands",
        "isActive": false,
        "title": "Pacific Islands",
      },
      Object {
        "id": "africa",
        "isActive": false,
        "title": "Africa",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  types={
    Array [
      Object {
        "id": "temperature",
        "isActive": true,
        "title": "Temperature",
      },
      Object {
        "id": "precipitation",
        "isActive": false,
        "title": "Precipitation",
      },
      Object {
        "id": "pressure",
        "isActive": false,
        "title": "Pressure",
      },
      Object {
        "id": "wind",
        "isActive": false,
        "title": "Wind",
      },
      Object {
        "id": "cloud",
        "isActive": false,
        "title": "Cloud",
      },
    ]
  }
/>
`;
